src/nvme/fabrics.c

```diff
@@ -571,13 +568,13 @@ int nvmf_add_ctrl_opts(nvme_ctrl_t c, struct nvme_fabrics_config *cfg)
 		return -1;
 	}
 
-	ret = __nvmf_add_ctrl(argstr);
+	ret = __nvmf_add_ctrl(c, argstr);
 	free(argstr);
 	if (ret < 0) {
 		errno = -ret;
 		ret = -1;
 	} else {
-		nvme_msg(LOG_INFO, "nvme%d: ctrl connected\n", ret);
+		nvme_msg_n(c->s->h->r, LOG_INFO, "nvme%d: ctrl connected\n", ret);
```

Comment:
Indeed, looks a bit ugly. Though I don't have a good idea how to do it differently.